src/_nebari/stages/kubernetes_services/template/modules/kubernetes/services/jupyterhub/files/jupyterhub/03-profiles.py

```diff
@@ -97,22 +97,32 @@ def base_profile_shared_mounts(groups):
             {"name": "shared", "persistentVolumeClaim": {"claimName": shared_pvc_name}}
         )
 
+    relevant_groups = []
+    for group in groups:
+        for roles in group_role_mapping.get(group, []):
+            # Check if the group has a role that has a shared-directory scope
+            if "shared-directory" in roles.get("attributes", {}).get("resource", []):
```

Comment:
This key should be `component` (instead of `resource`) to be consistent with rest of roles and permissions for jupyterhub and conda-store. 

Example: https://github.com/nebari-dev/nebari/blob/4f8fc54189df896fdb2c74b0fb15516b34c660b0/src/_nebari/stages/kubernetes_services/template/modules/kubernetes/services/jupyterhub/files/jupyterhub/04-auth.py#L160